training/training.module

```diff
@@ -0,0 +1,479 @@
+    'arguments' => array('form' => NULL),
+  );
+
+  return $theme;
+}
+
+/**
+ * Theming a custom layout for user_login_block.
+ *
+ * @param array $form
+ *   Array of form elements.
+ *
+ * @return string
+ *  String contents HTML tags for custom layout.
+ */
+function theme_test_form_theme_function($form) {
+  $attributes = array(
+    'id' => 'table-test-login',
+  );
+  $rows = array(
+    array(
+      drupal_render($form['form']['name']),
+    ),
+    array(
+      drupal_render($form['form']['pass']),
+    ),
+    array(
+      drupal_render($form['form']['secret_key']),
+    ),
+    array(
+      drupal_render($form['form']['actions']['submit']),
+    ),
+    array(
+      drupal_render($form['form']['links']) .
+      drupal_render($form['form']['form_build_id']) .
+      drupal_render($form['form']['form_id']),
+    ),
+  );
+
+  return theme('table', array(
+    'rows' => $rows,
+    'attributes' => $attributes,
+  ));
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function training_block_info() {
+  $blocks['exchange'] = array(
+    'info' => t('Exchange'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ *
+ * Sends a request to the specified address. Parses the xml data. Generates
+content for the block.
+ */
+function training_block_view($delta='') {
+  $date = '';
+  $content = '';
+  $request = 'http://www.nbrb.by/Services/XmlExRates.aspx';
+  $request = drupal_http_request($request);
+  $data = new SimpleXMLElement($request->data);
+  foreach ($data->attributes() as $key => $val) {
+    $date .= $val;
+  }
+  foreach ($data->Currency as $value) {
+    if (variable_get('check' . $value->CharCode)) {
+      $content .= '<tr><td>' .  $value->Name . '</td>
+                   <td>' . $value->CharCode . '</td>
+                   <td>' . $value->Rate . '</td></tr>';
+    }
+  }
+  $block = array();
+  if ($delta == 'exchange') {
+    $block['subject'] = t('Title block');
+    $block['content'] = t('Обновлено %date', array('%date' => $date)) .
+                          '<table>' . $content . '</table>';
+  }
+
+  return $block;
+}
+
+/**
+ * Implements hook_block_configure().
+ */
+function training_block_configure($delta = '') {
+  $form = array();
+  if ($delta == 'exchange') {
+    $request = 'http://www.nbrb.by/Services/XmlExRates.aspx';
+    $request = drupal_http_request($request);
+    $data = new SimpleXMLElement($request->data);
+    foreach ($data->Currency as $value) {
+      $form['check' . $value->CharCode] = array(
+        '#type' => 'checkbox',
+        '#title' => $value->Name,
+        '#default_value' => variable_get('check' . $value->CharCode),
+      );
+    }
+  }
+
+  return $form;
+}
+
+/**
+ * Implements hook_block_save().
+ */
+function training_block_save($delta = '', $edit = array()) {
+  if ($delta == 'exchange') {
+    $request = 'http://www.nbrb.by/Services/XmlExRates.aspx';
+    $request = drupal_http_request($request);
```

Comment:
Use DRY principle of software programming 
don’t repeat yourself (DRY) - http://en.wikipedia.org/wiki/Don't_repeat_yourself
use constant for $request = 'http://www.nbrb.by/Services/XmlExRates.aspx';